3-typescript/1-ecommerce/2-products.ts

```diff
@@ -35,13 +95,48 @@ async function analyzeProductPrices(products: any[]): Promise<any> {}
   - The brandInfo field should include the rest of the brand metadata (name, logo, description, etc.).
  */
 
-async function buildProductCatalog(
-  products: unknown[],
-  brands: unknown[],
-): Promise<unknown[]> {
-  return [];
+interface EnrichedProduct extends Product {
+  brandInfo: Brand;
 }
 
+export async function buildProductCatalog(
+  products: Product[],
+  brands: Brand[],
+): Promise<EnrichedProduct[]> {
+  let enrichedProducts: EnrichedProduct[] = [];
+  products.map((product) => {
+    let enrichedProduct: EnrichedProduct;
+    if (!product.isActive) {
+      return;
+    }
+
+    const brand = brands.find(
+      (brand) => brand.id === product.brandId && brand.isActive,
+    );
+
+    if (brand === undefined || !brand.isActive) {
```

Comment:
Since brand is expected to be an object you can leverage on falsy values such as null, undefined.
```suggestion
    if (!brand  || !brand.isActive) {
```